pasta/augment/inline.py

```diff
@@ -0,0 +1,66 @@
+# coding=utf-8
+"""Inline constants in a python module."""
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import ast
+import copy
+import itertools
+import six
+
+from pasta.augment import import_utils
+from pasta.base import ast_utils
+from pasta.base import scope
+
+
+class InlineError(Exception):
+  pass
+
+
+def inline_name(t, name):
+  """Inline a constant name into a module."""
+  sc = scope.analyze(t)
+  name_node = sc.names[name]
+
+  # The name must be a Name node (not a FunctionDef, etc.)
+  if not isinstance(name_node.definition, ast.Name):
+    raise InlineError('%r is not a constant; it has type %r' % (
+        name, type(name_node.definition)))
+
+  assign_node = sc.parent(name_node.definition)
```

Comment:
What if this isn't an assign node? For example:

* `with x as a: pass` will have a `With` parent
* `(a, b) = x`, `a` will have a `Tuple` parent 

Can you write test cases for these? I think they'll raise an AttributeError on the next line, but I'd prefer `InlineError`.

Reply:
You're right I'm not handling these cases right now, but they are not something I would want to inline anyway (probably). Will add more error checking + tests to this pull though.